.travis.yml

```diff
@@ -1,28 +1,32 @@
-# ensure running in a container for fast boot
+# ensure running in a container for fast boot, travis has these requirements:
 sudo: false
 dist: "trusty"
 
-# these are executed in order.  each must pass for the next to be run
+# these are executed in order. each must pass for the next to be run
 stages:
   - precache # warm up cache for default Node.js version
   - lint # lint code and docs
   - test # all tests
 
-#defaults
+# defaults
 language: node_js
-node_js: "node"
+node_js: "node" # equivalent to stable, node 10
+
 # `nvm install` happens before the cache is restored, which means
-# we must install our own npm elsewhere (`~/npm`)
+# we must install our own npm elsewhere (`~/cached-bins`)
 before_install: |
+  # Evaluate with Bash if the npm in cached_bins path has NOT execution permission,
   [[ ! -x ~/cached-bins/node_modules/.bin/npm ]] && {
-    # caching feature creates `~/global-bins` for us
+  # then we assume cache missed and install npm@latest and lerna
+  # caching feature creates `~/cached-bins` for us, the private:true is to avoid annoying npm warnings about package.json
     cd ~/cached-bins && echo '{"private":true}' > package.json && npm install npm@latest lerna --no-package-lock
     cd -
   } || true
   # avoids bugs around https://github.com/travis-ci/travis-ci/issues/5092
   export PATH=~/cached-bins/node_modules/.bin:$PATH
-# this avoids compilation in most cases (where we don't need it)

```

Comment:
In our case this does not avoid compilation since we use native modules for some crypto libraries, and those must be compiled